Shared/GeometryQuadtree.cpp

```diff
@@ -150,9 +151,9 @@ QList<Geometry> GeometryQuadtree::candidateIntersections(const Envelope& extent)
     auto findIt = m_elementStorage.find(id);
     if (findIt != m_elementStorage.end())
     {
-      GeoElement* element = findIt.value();
+      GeoElementSignaler* element = findIt.value();
       if (element)
-        results.push_back(element->geometry());
+        results.push_back(element->m_geoElement->geometry());
```

Comment:
@JamesMBallard In this case, is the `GeoElementSignaler` required or can the code here remain untouched?  There are others in this PR that are similar.

Reply:
I need to persist the object with the signal so we can listen for it. Unfortunately that means we need change the member to this. Not too bad and the `GeoElementSignaler` just wraps the geoElement.

```cpp
QHash<int, GeoElementSignaler*> m_elementStorage;
```

Reply:
@JamesMBallard But it looks like for this particular code that `GeoElement` wasn't being persisted before these changes.  It looks like we're just pulling out the current geometry and hooking up to the `geometryChanged` signal.  I assume that's being done elsewhere.  Let's chat offline if I'm missing something obvious.

Reply:
@michael-tims the GeoElement was being persisted before in the `m_elementStorage` member. This local `element` is being pulled out of `m_elementStorage`.